src/docker/start.sh

```diff
@@ -1,20 +1,22 @@
-#!/bin/sh
-
-if ! [ -z "$CONF_FILE_PATH" ] && ! [ -f "CONF_FILE_PATH" ]; then
+#!/usr/bin/env bash
+if [[ -n "$CONF_FILE_PATH" ]] && [ ! -f "CONF_FILE_PATH" ]; then
     echo "Generating config file $CONF_FILE_PATH"
     touch "CONF_FILE_PATH"
 
-    if ! [ -z "$MATRIX_DOMAIN" ]; then
+    if [[ -n "$MATRIX_DOMAIN" ]] \
+      && ! egrep -q "^matrix.domain:" $CONF_FILE_PATH; then
```

Comment:
This should not be present if the above `if` works properly: nothing should be done on the config file if it already exists.